src/Compilers/CSharp/Portable/Utilities/ValueSetFactory.ULongTC.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    using static BinaryOperatorKind;
+
+    internal static partial class ValueSetFactory
+    {
+        private struct ULongTC : NumericTC<ulong>
```

Comment:
>ULongTC [](start = 23, length = 7)

Looking at tests in `ValueSetTests.cs`, I only see two assertions related to `ForLong` (and therefore `ULongTC`). Where are we getting coverage for logic below?

Reply:
Those two tests actually do cover all the logic below through the use of the value sets.  Note that this logic only implements the basic operations for the long type.

---
In reply to: [381023530](https://github.com/dotnet/roslyn/pull/41503#discussion_r381023530) [](ancestors = 381023530)